crypto/src/symmetric.rs

```diff
@@ -0,0 +1,59 @@
+use anyhow::Result;
+use chacha20poly1305::{
+    aead::{Aead, NewAead},
+    ChaCha20Poly1305, Key, Nonce,
+};
+use once_cell::sync::Lazy;
+
+use crate::ka;
+
+/// The nonce used for note encryption.
+pub static NOTE_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| [0u8; 12]);
+
+/// The nonce used for memo encryption.
+pub static MEMO_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| {
+    let nonce_bytes = 1u128.to_le_bytes();
+    nonce_bytes[0..12].try_into().expect("nonce fits in array")
+});
+
+/// The nonce used for swap encryption.
+pub static SWAP_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| [9u8; 12]);
```

Comment:
Two comments here:

1. Because we're already specifying static `[u8; 12]` values, we're not really getting much from using a `Lazy` here, just overhead.  We could skip it by making the constant be of type `[u8; 12]` directly and writing out the arrays.
2. The current encryption/decryption API takes the nonce as bytes. But since we know that we want to be using specific nonces for specific types of data, I'm wondering if we should reshape the API to avoid the byte arrays entirely?

This could look like an
```rust
enum PayloadKind {
    Note,
    Memo,
    Swap,
}
```
with
```rust
impl PayloadKind {
    fn nonce(&self) -> [u8; 12] {
        match self {
            Self::Note => [0,0,0,0,0,0,0,0,0,0,0,0],
            Self::Memo => [1,0,0,0,0,0,0,0,0,0,0,0],
            Self::Swap => [2,0,0,0,0,0,0,0,0,0,0,0],
        }
    }
}
```
so that the `PayloadKey` methods can take a `PayloadKind` and call `kind.nonce()` internally to get the bytes.  Then callers don't need to know anything about the bytes, or import a bunch of different constants, they just need to have the `PayloadKind`.  It also makes the safety rule more coherent: each `PayloadKey` can only be used to encrypt at most one object of each kind.

Reply:
thanks for this suggestion! This is way better that what I was doing so I've added a `PayloadKind` into this PR